packages/core/content-manager/server/src/services/document-metadata.ts

```diff
@@ -79,26 +89,57 @@ export default ({ strapi }: { strapi: Core.Strapi }) => ({
     delete versionsByLocale[version.locale];
 
     // For each locale, get the ones with the same status
-    return (
-      Object.values(versionsByLocale)
-        .map((localeVersions: DocumentVersion[]) => {
-          // There will not be a draft and a version counterpart if the content type does not have draft and publish
-          if (!contentTypes.hasDraftAndPublish(strapi.getModel(uid))) {
-            return pick(AVAILABLE_LOCALES_FIELDS, localeVersions[0]);
+    // There will not be a draft and a version counterpart if the content
+    // type does not have draft and publish
+    const model = strapi.getModel(uid);
+    const keysToKeep = [...AVAILABLE_LOCALES_FIELDS, ...validatableFields];
```

Comment:
how does the validatableFields string[] works on nested schemas? what happens if the root has a "name" field, but a component inside also has a name field?

Reply:
I think this is covered in this test

https://github.com/strapi/strapi/blob/483791373ac11e4d33672122e16626f3104957e8/tests/api/core/content-manager/api/document-metadata.test.api.js#L266-L267

Both the product CT and features components have a 'name' string field

https://github.com/strapi/strapi/blob/483791373ac11e4d33672122e16626f3104957e8/packages/core/content-manager/server/src/services/document-metadata.ts#L230-L231
The populate here becomes 

```
	{
	  name: true,
	  category: true,
	  price: true,
	  features: {
	    populate: {
	    },
	  },
	}
```

The keys of which are used to build `keysToKeep`. So `getAvailableLocales` returns

```
	{
	  id: 2,
	  documentId: "bpd5tcr2mngn2j0e15jjakq9",
	  name: "prod-fr-draft",
	  category: "Cat-1-fr",
	  price: 1,
	  createdAt: "2024-05-10T11:23:53.441Z",
	  updatedAt: "2024-05-10T11:23:53.441Z",
	  publishedAt: null,
	  locale: "fr",
	  features: {
	    id: 2,
	    name: "Feature 1 fr",
	  },
	  status: "draft",
	}
```

Reply:
Probably not a big deal, but this might return more atteibutes than necessary. But I see!